tests/basic/upgrade_test.go

```diff
@@ -511,3 +512,198 @@ var _ = Describe("{UpgradeVolumeDriverFromCatalog}", func() {
 })
+
+var _ = Describe("{UpgradePxKvdbMemberDown}", func() {
+	/*
+					https://purestorage.atlassian.net/browse/PTX-21450
+				    https://portworx.testrail.net/index.php?/cases/view/94371
+					1. Bring down a kvdb member node
+			        2. Upgrade PX
+			        3. Bring up the kvdb member node
+		            4. Verify if the kvdb member joins the cluster and everything is running fine
+	*/
+
+	StartTorpedoTest("UpgradePxKvdbMemberDown", "Validating PX KVDB member down during upgrade", nil, 0)
+
+	var contexts []*scheduler.Context
+
+	var timeBeforeUpgrade time.Time
+	var timeAfterUpgrade time.Time
+	var kvdbNode node.Node
+
+	itLog := "Upgrade PX when kvdb member is down and after upgrade the kvdb member should join the kvdb cluster"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		storageNodes, err := GetStorageNodes()
+		log.FailOnError(err, "Failed to get storage nodes")
+		numOfNodes := len(storageNodes)
+
+		stepLog := "schedule applications"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				contexts = append(contexts, ScheduleApplications(fmt.Sprintf("upgradevolumedriver-%d", i))...)
+			}
+		})
+		ValidateApplications(contexts)
+		defer DestroyApps(contexts, nil)
+
+		if len(Inst().UpgradeStorageDriverEndpointList) == 0 {
+			log.Fatalf("Unable to perform volume driver upgrade hops, none were given")
+		}
+
+		for _, upgradeHop := range strings.Split(Inst().UpgradeStorageDriverEndpointList, ",") {
+			stepLog = "Add labels to non kvdb nodes so that they don't try to become kvdb node"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				for _, n := range storageNodes {
+					kvdbFlag, err := CheckIfNodeIsKvdbNode(n)
+					log.FailOnError(err, "Failed to check if node is kvdb node")
+					if !kvdbFlag {
+						err = Inst().S.AddLabelOnNode(n, k8s.MetaDataLabel, "false")
+						log.FailOnError(err, "Failed to add labels to node : %s", n.Name)
+						log.InfoD("Added label: %v to node: %s", k8s.MetaDataLabel, n.Name)
+					}
+				}
+			})
+
+			stepLog = "Bring down a kvdb member node"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				kvdbNodes, err := GetAllKvdbNodes()
+				log.FailOnError(err, "Failed to get kvdb nodes")
+
+				if len(kvdbNodes) < 2 {
+					log.FailOnError(fmt.Errorf("KVDB nodes are less than 2"), "KVDB nodes are less than 2")
+				}
+
+				kvdbNode, err = node.GetNodeDetailsByNodeID(kvdbNodes[rand.Intn(len(kvdbNodes))-1].ID)
+				log.FailOnError(err, "Failed to get kvdb node details")
+				log.InfoD("KVDB node to be stopped: %s", kvdbNode.Name)
+
+				err = Inst().V.StopDriver([]node.Node{kvdbNode}, false, nil)
+				log.FailOnError(err, "Failed to stop kvdb node")
+				log.InfoD("Stopped kvdb node: %s", kvdbNode.Name)
+			})
+
+			var wg sync.WaitGroup
+
+			stepLog = "Upgrade PX"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					currPXVersion, err := Inst().V.GetDriverVersionOnNode(storageNodes[0])
```

Comment:
why is this in go routine?

Reply:
So In this test case when upgrade is in progress kvdb member has to be down and a new member has to be selected for kvdb (when kvdb member is down the upgrade won't go through). so upgrade I have put in go routine and new member label removal and getting the old kvdb member up in the main go routine. that is why we need a go routiine to be there.